freva-client/src/freva_client/query.py

```diff
@@ -876,5 +937,205 @@ def _delete(self, url: str, **kwargs: Any) -> Optional[requests.models.Response]
             requests.exceptions.HTTPError,
         ) as error:
             msg = f"DELETE request failed with {error}"
+            if self._fail_on_error:
+                raise ValueError(msg) from None
             logger.warning(msg)
         return None
+
+    def _validate_user_data(
+        self,
+        user_data: Sequence[Union[str, xr.Dataset]],
+    ) -> Tuple[List[os.PathLike[str]], List[xr.Dataset]]:
+
+        validated_paths: List[os.PathLike[str]] = []
+        validated_xarray_datasets: List[xr.Dataset] = []
+
+        for data in user_data:
+            if isinstance(data, str):
+                path = Path(data)
+                if not path.exists():
```

Comment:
I think it would be kind of nice to have a functionality that can handle wild cards, so that something like this would work `~/data/*.nc`.